data-publisher/src/main/kotlin/no/nav/amt/tiltak/data_publisher/DataPublisherService.kt

```diff
@@ -0,0 +1,151 @@
+package no.nav.amt.tiltak.data_publisher
+
+import no.nav.amt.tiltak.common.json.JsonUtils
+import no.nav.amt.tiltak.data_publisher.model.DataPublishType
+import no.nav.amt.tiltak.data_publisher.publish.*
+import no.nav.amt.tiltak.kafka.config.KafkaTopicProperties
+import no.nav.common.kafka.producer.KafkaProducerClient
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.slf4j.LoggerFactory
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Service
+import java.util.*
+
+@Service
+class DataPublisherService(
+	private val kafkaTopicProperties: KafkaTopicProperties,
+	private val kafkaProducerClient: KafkaProducerClient<ByteArray, ByteArray>,
```

Comment:
Er det noen spesiell grunn til at du bruker ByteArray og ikke String? Jeg synes iallefall det er enklere å jobbe med String enn byte array :)

Reply:
Enig, men common biblioteket tar bare ByteArray, er hvertfall det som er brukt i resten av appen

Reply:
Dette synes jeg bør sjekkes litt nærmere. ByteArray er et mindre praktisk format å forholde seg til når det som sendes egentlig er en string. Når dataene ligger som string på topicen er det mye enklere både for appene som leser fra topicen, men også hvis vi må inn og se på hva som ligger på topicen for å sjekke en melding manuelt. Hvis common kun støtter byte array blir det såpass begrensende at jeg synes vi bør la være å bruke det og heller bygge opp dette med bare vanlig spring kafka.

Reply:
Ser ut som at du fritt kan velge typen selv i common-biblioteket: https://github.com/navikt/common-java-modules/blob/main/kafka/src/main/java/no/nav/common/kafka/producer/KafkaProducerClientImpl.java#L15